src/ferui-components/tree-view/tree-view-component.ts

```diff
@@ -0,0 +1,399 @@
+import { Observable, fromEvent, Subscription } from 'rxjs';
+import { throttleTime } from 'rxjs/internal/operators';
+import { FuiTreeNodeComponent } from './tree-node-component';
+
+@Component({
+  selector: 'fui-tree-view',
+  template: `
+    <div id="fui-tree-view-component" [ngStyle]="treeViewStyles" [ngClass]="colorTheme">
+      <fui-virtual-scroller
+        [hidden]="loading || error"
+        #scroll
+        class="fui-virtual-scroller"
+        [items]="scrollViewArray"
+        [bufferAmount]="BUFFER_AMOUNT"
+      >
+        <fui-tree-node
+          *ngFor="let node of scroll.viewPortItems"
+          [node]="node"
+          [theme]="colorTheme"
+          [borders]="hasBorders"
+          [dataRetriever]="dataRetriever"
+          (onNodeEvent)="nodeEvent($event)"
+        ></fui-tree-node>
+      </fui-virtual-scroller>
+      <clr-icon *ngIf="loading" class="fui-loader fui-loader-animation" shape="fui-spinner"></clr-icon>
+      <div *ngIf="error" class="error">
+        <clr-icon class="fui-error-icon" shape="fui-error" aria-hidden="true"></clr-icon>
+        <p>Couldn't load content</p>
+      </div>
+    </div>
+  `,
+  styles: [
+    `
+      #fui-tree-view-component {
+        padding: 10px;
+      }
+      .fui-loader {
+        color: #87a1b5;
+        position: relative;
+        top: 50%;
+        left: 50%;
+      }
+      .error {
+        position: relative;
+        top: 50%;
+        left: 50%;
+        color: #87a1b5;
+        text-align: center;
+        transform: translateX(-50%);
+      }
+      .DARK_BLUE {
+        background: #252a3a;
+        color: #fff;
+      }
+      .LIGHT_BLUE {
+        background: #03a6ff;
+        color: #fff;
+      }
+      .GRAY {
+        background: #f5f8f9;
+        color: #252a3a;
+      }
+      .WHITE {
+        background: #fff;
+        color: #252a3a;
+      }
+      .fui-virtual-scroller {
+        height: 100%;
+        width: 100%;
+      }
+      :host(.show-border) #fui-tree-view-component {
+        border: #e5eaee 1px solid;
+        padding: 0;
+      }
+    `,
+  ],
+})
+export class FuiTreeViewComponent<T> implements OnInit, OnDestroy {
+  @Input() treeNodeData: TreeNodeData<T>;
+
+  @Input() dataRetriever: TreeNodeDataRetriever<T> | PagedTreeNodeDataRetriever<T>;
+
+  @Input() config: TreeViewConfiguration;
+
+  @Input() loading: boolean = false;
+
+  @Input() error: boolean = false;
+
+  @HostBinding('class.show-border') border: boolean;
+
+  @ViewChild(FuiVirtualScrollerComponent) vs: FuiVirtualScrollerComponent;
+
+  @ViewChildren(FuiTreeNodeComponent) children: QueryList<FuiTreeNodeComponent<T>>;
+
+  @Output() onNodeEvent: EventEmitter<TreeViewEvent<any>> = new EventEmitter<TreeViewEvent<any>>();
+
+  treeViewStyles: FuiTreeViewComponentStyles;
+  colorTheme: TreeViewColorTheme;
+  scrollViewArray: TreeNode<T>[];
+  hasBorders: boolean;
+  BUFFER_AMOUNT: number = 1000;
```

Comment:
Why your buffer amount is that high by default ? You should keep it as small as it can be. A good default value would be something like 5 or 10. Remember that it doesn't correspond to the number of element visible in virtual-scroller but to the number of extra html element we will load in DOM to insure a smooth scroll. It is not optimized to set it to 1000 by default... The virtual scroller comes almost useless in that case.

Reply:
I agree, this buffer default was set by Cory. @cfsnyder do you agree we should lower this? 
I think 100 should be a good enough buffer

Reply:
The buffer amount that I was arguing could be 1000 was the amount that we preload from the server. If this is also used as the configuration parameter for the number of extra items that we load in the DOM for the virtual scroller, I think the two configuration params should be separated. I agree that the one @JoMas971  is referring to should be much lower.